src/authn/ccloudStateHandling.test.ts

```diff
@@ -6,25 +6,26 @@ import { TEST_CCLOUD_CONNECTION } from "../../tests/unit/testResources/connectio
 import { getTestExtensionContext } from "../../tests/unit/testUtils";
 import { Connection, Status } from "../clients/sidecar";
 import { nonInvalidTokenStatus } from "../emitters";
-import * as connections from "../sidecar/connections";
 import { ResourceManager } from "../storage/resourceManager";
 import {
   AuthPromptTracker,
   checkAuthExpiration,
   MINUTES_UNTIL_REAUTH_WARNING,
-  pollCCloudConnectionAuth,
+  reactToCCloudAuthState,
   REAUTH_BUTTON_TEXT,
   REMIND_BUTTON_TEXT,
-  watchCCloudConnectionStatus,
-} from "./ccloudPolling";
+} from "./ccloudStateHandling";
 
 configDotenv();
 
 /**
  * Light test wrapper to update the `requires_authentication_at` field in a {@link Connection} object.
  */
 function createFakeConnection(expiresInMinutes: number | undefined): Connection {
-  const connection = TEST_CCLOUD_CONNECTION;
+  // Make a deep clone of TEST_CCLOUD_CONNECTION, then mutate its authentication portion
```

Comment:
Fixes some suite flakiness I encountered when running whole suite. Can't change `TEST_CCLOUD_CONNECTION`'s state from being completely not logged in, otherwise hilarity sometimes ensues in tests that expect that.